orchagent/aclorch.cpp

```diff
@@ -2120,140 +2175,40 @@ void AclOrch::doAclRuleTask(Consumer &consumer)
-        string port_alias = key.substr(0, found);
-        string op = kfvOp(t);
-
-        SWSS_LOG_INFO("doAclTablePortUpdateTask: OP: %s, port_alias: %s", op.c_str(), port_alias.c_str());
-
-        if (op == SET_COMMAND)
-        {
-            for (auto itmap : m_AclTables)
-            {
-                auto table = itmap.second;
-                if (table.pendingPortSet.find(port_alias) != table.pendingPortSet.end())
-                {
-                    SWSS_LOG_INFO("found the port: %s in ACL table: %s pending port list, bind it to ACL table.", port_alias.c_str(), table.description.c_str());
-
-                    bool suc = processPendingPort(table, port_alias, [&](sai_object_id_t portOid) {
-                        table.link(portOid);
-                    });
+    auto port_list = tokenize(portList, ',');
+    set<string> ports(port_list.begin(), port_list.end());
 
-                    if (!suc)
-                    {
-                        SWSS_LOG_ERROR("Failed to bind the ACL table: %s to port: %s", table.description.c_str(), port_alias.c_str());
-                    }
-                    else
-                    {
-                        table.pendingPortSet.erase(port_alias);
-                        SWSS_LOG_DEBUG("port: %s bound to ACL table table: %s, remove it from pending list", port_alias.c_str(), table.description.c_str());
-                    }
-                }
-            }
-        }
-        else if (op == DEL_COMMAND)
-        {
-            for (auto itmap : m_AclTables)
-            {
-                auto table = itmap.second;
-                if (table.portSet.find(port_alias) != table.portSet.end())
-                {
-                    /*TODO: update the ACL table after port/lag deleted*/
-                    table.pendingPortSet.emplace(port_alias);
-                    SWSS_LOG_INFO("Add deleted port: %s to the pending list of ACL table: %s", port_alias.c_str(), table.description.c_str());
-                }
-            }
-        }
-        else
-        {
-            SWSS_LOG_ERROR("Unknown operation type %s", op.c_str());
-        }
-        it = consumer.m_toSync.erase(it);
-    }
-}
-
-bool AclOrch::processPorts(AclTable &aclTable, string portsList, std::function<void (sai_object_id_t)> inserter)
-{
-    SWSS_LOG_ENTER();
-
-    vector<string> strList;
-
-    SWSS_LOG_DEBUG("Processing ACL table port list %s", portsList.c_str());
-
-    split(portsList, strList, ',');
-
-    set<string> strSet(strList.begin(), strList.end());
-    aclTable.portSet = strSet;
-
-    if (strList.size() != strSet.size())
-    {
-        SWSS_LOG_ERROR("Failed to process port list. Duplicate port entry");
-        return false;
-    }
-
-    if (strList.empty())
+    if (ports.empty())
     {
-        SWSS_LOG_ERROR("Failed to process port list. List is empty");
+        SWSS_LOG_ERROR("Failed to process empty port list");
```

Comment:
>SWSS_LOG_ERROR [](start = 8, length = 14)

why error level? it mean the acl table currently does not bind to any port. It should be a valid configuration. We probably should not print out anything.

Reply:
Right now adding ports is not supported when creating the ACL table. This log means that the ACL table entry doesn't have any port associated with this. If we want to add ports to this ACL table, the current behavior is to remove old ACL table and create a new one. I have added TODO here.